### PR TITLE
Adding acoustic subsystem examples

### PR DESCRIPTION
**Pull request by marcoag**
Adding the acoustic subsystem examples as suggest in https://github.com/osrf/lrauv/issues/224#issuecomment-1214223663.

The related documentation is hosted in the [wiki](https://github.com/osrf/lrauv/wiki/Examples-and-tutorials#how-to-send-and-receive-data-packets-on-the-acoustic-subsystem).

Signed-off-by: Marco A. Gutierrez <marco@openrobotics.org>

### CODE REVIEW

File: lrauv_ignition_plugins/example/example_comms_echo.cc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+{
+  transmitter = node.Advertise<lrauv_ignition_plugins::msgs::LRAUVAcousticMessage>(
+    "/comms/external/" + std::to_string(address) + "/tx");
+
+  node.Subscribe(
+    "/comms/external/" + std::to_string(address) + "/rx",
+    cb
+  );
+
+  while(true) { sleep(1); }
```
Comment (hidmic) [inline lrauv_ignition_plugins/example/example_comms_echo.cc:74]: @marcoag IIRC there's a `waitForShutdown` API in `gz::transport`.

File: lrauv_ignition_plugins/example/example_comms_echo.cc [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ *   $ ./LRAUV_example_comms_echo
+ */
+
+#include <unistd.h>
+#include <gz/transport/Node.hh>
+#include "lrauv_ignition_plugins/lrauv_acoustic_message.pb.h"
+
+int address = 1;
+gz::transport::Node node;
+gz::transport::Node::Publisher transmitter;
```
Comment (hidmic) [inline lrauv_ignition_plugins/example/example_comms_echo.cc:38]: @marcoag perhaps we move these to the `main` function below? IIRC `gz::transport` allows for lambdas as subscription callbacks.

File: lrauv_ignition_plugins/example/example_comms_echo.cc [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * Usage:
+ *   $ ./LRAUV_example_comms_echo
+ */
+
+#include <unistd.h>
+#include <gz/transport/Node.hh>
+#include "lrauv_ignition_plugins/lrauv_acoustic_message.pb.h"
+
+int address = 1;
```
Comment (hidmic) [inline lrauv_ignition_plugins/example/example_comms_echo.cc:36]: @marcoag nit: would it make sense for the address to be a CLI argument?

File: lrauv_ignition_plugins/example/example_comms_echo.cc [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+/*
+ * Development of this module has been funded by the Monterey Bay Aquarium
+ * Research Institute (MBARI) and the David and Lucile Packard Foundation
+ */
+
+/**
+ * This is a small example of usage of gazebo transport to send and receive
+ * data packets using the acoustic subsystem. For more info:
+ * https://github.com/osrf/lrauv/wiki/Examples-and-tutorials#direct-method
```
Comment (hidmic) [inline lrauv_ignition_plugins/example/example_comms_echo.cc:26]: @marcoag qq, will there a be instructions on how to use these examples in the wiki? If not, perhaps it's worth adding them here.

File: lrauv_ignition_plugins/example/example_comms_client.cc [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  // Who to send to
+  msg.set_to(2);
+  // From who
+  msg.set_from(1);
+  // `LRAUVAcousticMessage_MessageType_Other` means its a data packet
+  msg.set_type(AcousticMsg::MessageType::LRAUVAcousticMessage_MessageType_Other);
+  // The data
+  msg.set_data("test_message");
+  client.SendPacket(msg);
+
```
Comment (hidmic) [inline lrauv_ignition_plugins/example/example_comms_client.cc:60]: @marcoag nit: unnecessary blank line.

File: lrauv_ignition_plugins/example/example_comms_client.cc [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+ */
+
+#include <lrauv_ignition_plugins/comms/CommsClient.hh>
+
+using namespace tethys;
+using AcousticMsg = lrauv_ignition_plugins::msgs::LRAUVAcousticMessage;
+
+int main(int _argc, char **_argv)
+{
+
```
Comment (hidmic) [inline lrauv_ignition_plugins/example/example_comms_client.cc:39]: @marcoag nit: unnecessary blank line.

Comment (marcoag) [reply lrauv_ignition_plugins/example/example_comms_client.cc]: Removed.

File: lrauv_ignition_plugins/example/example_comms_client.cc (hunk 5 above)
Comment (marcoag) [reply lrauv_ignition_plugins/example/example_comms_client.cc]: Removed.

File: lrauv_ignition_plugins/example/example_comms_echo.cc (hunk 4 above)
Comment (marcoag) [reply lrauv_ignition_plugins/example/example_comms_echo.cc:26]: There's a line that says how to run the example. Further documentation should be on the wiki, I guess it's better to have just one main source of documentation.

File: lrauv_ignition_plugins/example/example_comms_echo.cc (hunk 3 above)
Comment (marcoag) [reply lrauv_ignition_plugins/example/example_comms_echo.cc:37]: Added an option to pass a CLI argument for this.

File: lrauv_ignition_plugins/example/example_comms_echo.cc (hunk 1 above)
Comment (marcoag) [reply lrauv_ignition_plugins/example/example_comms_echo.cc]: Good point, added.

**Review by hidmic — COMMENTED** [review]
LGTM but for one last comment

File: lrauv_ignition_plugins/example/example_comms_client.cc [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  AcousticMsg msg;
+  // Who to send to
+  msg.set_to(2);
+  // From who
+  msg.set_from(1);
+  // `LRAUVAcousticMessage_MessageType_Other` means its a data packet
+  msg.set_type(AcousticMsg::MessageType::LRAUVAcousticMessage_MessageType_Other);
+  // The data
+  msg.set_data("test_message");
+  client.SendPacket(msg);
```
[the remark this answers is not in the document]
Comment (marcoag) [reply lrauv_ignition_plugins/example/example_comms_client.cc:58]: tbh, I'm not sure... but could be? I mostly copied whatever was there on the documentation example before.

[the remark this answers is not in the document]
Comment (hidmic) [reply lrauv_ignition_plugins/example/example_comms_client.cc:58]: @marcoag could we wait for the response to arrive before exiting?

[the remark this answers is not in the document]
Comment (marcoag) [reply lrauv_ignition_plugins/example/example_comms_client.cc]: Sure, just added a sender and a receiver for extra clarity and some more verbosity to the example.

**Review by hidmic — APPROVED** [review]

**Pull request merged**